clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractHeartbeatRequestManager.java

```diff
@@ -461,15 +434,24 @@ protected void handleFatalFailure(Throwable error) {
         membershipManager().transitionToFatal();
     }
 
+    /**
+     * Error handling specific failure to a group type when sending the request.
```

Comment:
I would add "and no response has been received" to the comment. This is the crucial difference between the usage of `handleSpecificFailure` and `handleSpecificExceptionInResponse`.